### PR TITLE
Add API to reinitialize VSC7448

### PR DESCRIPTION
**Pull request by mkeeter**
This is a potential workaround for bf_sde#27; the idea would be to reset the entire Monorail switch + PHYs if the 10G link went down persistently.

I'd like to see it in action before installing it automatically as a watchdog, so this PR just adds an API endpoint.  Next time we encounter the issue in the wild, we can hit the endpoint and see if things resolve.

In addition, there's a bit of cleanup:

- Moves disabling `VAUI_CHANNEL_CFG.LANE_SYNC_ENA` to _before_ the VSC8504 is brought up.  In benchtop testing, this seems to fix https://github.com/oxidecomputer/hubris/issues/1048 , which is _plausible_ if not obvious.
- Make `Monorail` API use `drv_monorail_api::*` types, to avoid an ambiguous decoding in Humility
- Remove the Monorail BSP for the VSC7448 dev kit; the `app-vsc7448.toml` was removed a while back so this is dead code (🪦)

### CODE REVIEW

File: drv/vsc7448/src/lib.rs [hunk 1]
```diff
@@ -366,11 +376,7 @@ impl<'a, R: Vsc7448Rw> Vsc7448<'a, R> {
     ///
     /// Takes the REFCLK frequency, as well as an optional frequency for
     /// REFCLK2 (used to configure the PLL boost).
```
Comment (jgallagher) [inline drv/vsc7448/src/lib.rs:378]: This part of the comment should presumably move to `new()`?

File: task/monorail-server/src/bsp/sidecar_bc.rs [hunk 2]
```diff
@@ -299,6 +309,7 @@ impl<'a, R: Vsc7448Rw> Bsp<'a, R> {
         // on GPIO_47.
         const COMA_MODE_GPIO: u32 = 47;
 
+        //
```
Comment (jgallagher) [inline task/monorail-server/src/bsp/sidecar_bc.rs:312]: Stray addition or intentional formatting?

**Review by jgallagher — APPROVED** [review]

File: task/monorail-server/src/bsp/sidecar_bc.rs (hunk 2 above)
Comment (mkeeter) [reply task/monorail-server/src/bsp/sidecar_bc.rs:312]: Stray addition, fixed!

File: drv/vsc7448/src/lib.rs (hunk 1 above)
Comment (mkeeter) [reply drv/vsc7448/src/lib.rs]: Yup, done

**Pull request merged**